pinot-core/src/main/java/org/apache/pinot/core/operator/filter/RangeIndexBasedFilterOperator.java

```diff
@@ -44,44 +53,74 @@ public class RangeIndexBasedFilterOperator extends BaseFilterOperator {
 
   private static final String EXPLAIN_NAME = "FILTER_RANGE_INDEX";
 
-  private final RangeEvaluator _rangeEvaluator;
-  private final PredicateEvaluator _rangePredicateEvaluator;
+  static final Set<FieldSpec.DataType> SUPPORTED_RAW_DATA_TYPES =
+      EnumSet.of(FieldSpec.DataType.INT, FieldSpec.DataType.LONG, FieldSpec.DataType.FLOAT, FieldSpec.DataType.DOUBLE);
+
+  private final RangeIndexReader<ImmutableRoaringBitmap> _rangeIndexReader;
+  private final PredicateEvaluator _predicateEvaluator;
   private final DataSource _dataSource;
+  private final FieldSpec.DataType _parameterType;
+  private final Object _parameters;
   private final int _numDocs;
 
+  static boolean canEvaluate(PredicateEvaluator predicateEvaluator, DataSource dataSource) {
+    if (dataSource.getRangeIndex() != null) {
+      boolean datatypeSupported = (predicateEvaluator.isDictionaryBased() || SUPPORTED_RAW_DATA_TYPES.contains(
+          predicateEvaluator.getDataType()));
+      switch (predicateEvaluator.getPredicateType()) {
+        case EQ:
+          return datatypeSupported && dataSource.getRangeIndex().isExact() && dataSource.getRangeIndex().isExact() && (
+              predicateEvaluator instanceof DictionaryBasedEqPredicateEvaluator
+                  || predicateEvaluator instanceof IntRawValueBasedEqPredicateEvaluator
+                  || predicateEvaluator instanceof LongRawValueBasedEqPredicateEvaluator
+                  || predicateEvaluator instanceof FloatRawValueBasedEqPredicateEvaluator
+                  || predicateEvaluator instanceof DoubleRawValueBasedEqPredicateEvaluator);
+        case RANGE:
+          return datatypeSupported && (predicateEvaluator instanceof SortedDictionaryBasedRangePredicateEvaluator
+              || predicateEvaluator instanceof IntRawValueBasedRangePredicateEvaluator
+              || predicateEvaluator instanceof LongRawValueBasedRangePredicateEvaluator
+              || predicateEvaluator instanceof FloatRawValueBasedRangePredicateEvaluator
+              || predicateEvaluator instanceof DoubleRawValueBasedRangePredicateEvaluator);
+        default:
```

Comment:
(minor) add a `return false` for readability

Reply:
this code has been removed.